src/material.h

```diff
@@ -29,6 +30,9 @@ class material_type
         int _fire_resist;
         int _chip_resist;       // Resistance to physical damage of the item itself
         int _density;   // relative to "powder", which is 1
+        bool _edible;
+
+        mutable std::map<vitamin_id, double> _vitamins;
```

Comment:
> Materials aren't good for nutritional details.

See the earlier discussion including the example for deluxe scrambled egg. It's an excellent starting point as it produces reasonable values without the need to specify `vitamins` for each `FOOD` item (currently 471). I do want to add a `vitamins` field to `COMESTIBLE` to allow overriding this for any identified corner cases but this is waiting on #16136 to avoid a merge conflict and also so that foods can be broken into classes reducing the workload (fruit, veg etc).

That said the vitamins implementation is intended to be coarse. Fruit should contain a lot of vitamin C, vegetables not so much and junk food none. Cooked food should contain a mix of vitamins from the components. I want to avoid definitions of the type `apple: 2.9mg`, `pear: 3.1mg` because it wouldn't further the goal of preventing single source diets any better than coarser units.